subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -1179,13 +1188,34 @@ static inline void ticker_job_node_manage(struct ticker_instance *instance,
 		ticker->req = ticker->ack;
 
 		if (instance->ticker_id_slot_previous == user_op->id) {
-			u32_t ticks_now = cntr_cnt_get();
+			u32_t ticks_current;
+			u32_t ticks_at_stop;
 			u32_t ticks_used;
 
 			instance->ticker_id_slot_previous = TICKER_NULL;
-			ticks_used = ticks_elapsed +
-				ticker_ticks_diff_get(ticks_now,
-						      instance->ticks_current);
+			if (user_op->op == TICKER_USER_OP_TYPE_STOP_ABS) {
+				ticks_at_stop =
+					user_op->params.stop.ticks_at_stop;
+			} else {
+				ticks_at_stop = cntr_cnt_get();
+			}
+
+			ticks_current = instance->ticks_current;
+			if (!((ticks_at_stop - ticks_current) &
+			      BIT(HAL_TICKER_CNTR_MSBIT))) {
+				ticks_used = ticks_elapsed +
+					ticker_ticks_diff_get(ticks_at_stop,
+							      ticks_current);
+			} else {
+				ticks_used =
+					ticker_ticks_diff_get(ticks_current,
+							      ticks_at_stop);
+				if (ticks_used > ticks_elapsed) {
+					ticks_used -= ticks_elapsed;
+				} else {
+					ticks_used = 0;
+				}
```

Comment:
ok. Let me record my thought here.

`ticks_elapsed` is units if time over and above the currently ticker module aware absolute tick, `instance->ticks_current` (`ticks_current`).
Now, if the absolute `ticks_at_stop` is before `ticks_current`, then the negative ticks used is stored in `ticks_used`

Definitely, this block is incorrect.

Reply:
I think your latest commit is correct, but I will dig into the code some more later today.

Reply:
@mtpr-ot  Please notice the replaced MIN with a if-clause, this is a fix that may be needed upstream for split controller in v2.2 release. If you agree, we need to create a GH issue and fix it for the release.

Reply:
@cvinayak I can see your point. In the current 2.2 code, a counter wrap would cause an incorrect  `ticks_slot_previous`, but still MIN'ed to the normal `ticks_slot`. This would mean that a stopped node would "reserve" the full slot. It's not a serious problem, as I see it, but should be fixed. Do you agee?

Reply:
In the upstream code, `ticks_slot_previous` is reduced every time there is any other ticker with or without reservation (collision resolution) expiring within the current reserved space.

>  a stopped node would "reserve" the full slot. It's not a serious problem, as I see it, but should be fixed.

Agree.